lib/modules/versioning/deb/index.ts

```diff
@@ -0,0 +1,147 @@
+import { regEx } from '../../../util/regex';
+import { GenericVersion, GenericVersioningApi } from '../generic';
+import type { VersioningApi } from '../types';
+
+export const id = 'deb';
+export const displayName = 'Deb version';
+export const urls = [
+  'https://www.debian.org/doc/debian-policy/ch-controlfields.html#version',
+  'https://manpages.debian.org/unstable/dpkg-dev/deb-version.7.en.html',
+];
+export const supportsRanges = false;
+
+const upstreamVersionPattern = regEx(/^[-+.:~A-Za-z0-9]+$/);
+const debianRevisionPattern = regEx(/^[+.~A-Za-z0-9]*$/);
+const numericPattern = regEx(/[0-9]+/g);
+const characterOrder =
+  '~ ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz+-.:';
+const numericChars = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9'];
+
+export interface DebVersion extends GenericVersion {
+  /**
+   * epoch, defaults to 0 if not present, are used to leave version mistakes and previous
+   * versioning schemes behind.
+   */
+  epoch: number;
+  /**
+   * upstreamVersion is the main version part: it defines the version of origin software
+   * that was packaged.
+   */
+  upstreamVersion: string;
+  /**
+   * debianRevsion is used to distinguish between different versions of packaging for the
+   * same upstream version.
+   */
+  debianRevision: string;
+}
+
+class DebVersioningApi extends GenericVersioningApi {
+  protected _parse(version: string): DebVersion | null {
+    /* Splitting version into "[epoch:]upstream-version[-debian-revision]"
+       All found numbers are exported as release info */
+    let epoch = 0;
+    let nonEpochVersion = version;
+    if (nonEpochVersion.includes(':')) {
+      const epochEnd = nonEpochVersion.indexOf(':');
+      epoch = Number(nonEpochVersion.substring(0, epochEnd));
```

Comment:
use `parseInt(str, 10)`

Reply:
The conversion is also used to verify the epoch value is valid (unsigned integer). `parseInt` ignores not-parsable characters like dots (in `1.1`).

Should I stick to `Number` or first use a regEx check to verify the format is what we want?

(I added two more tests for these cases).

Reply:
I thought epoch can only be an unsigned Integer? so parseint fails for all non signed int.
so please use safer parseint and validate return value. you can also use regex to validate more, so `-1.54:...`. will be rejected right before further parsing.

Reply:
> so parseint fails for all non signed int.

This is not the case: `parseInt("1.1", 10) === 1`. In my tests and the unit tests, `parseInt` ignores trailing non-digit characters.